From 94df28da10df6e1e458bbb60e0c433c394d4df3c Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 28 Mar 2017 15:27:21 +0300
Subject: [PATCH 0929/1345] ahci: iommu: dts: cp110: add stream ID
 configuration for SATA unit

commit  010e5352326268dba851528ab271a7257f3f8ec7 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

A stream ID is a unique number that identifies the device
when it requires translation on the SMMU (MMU-500).
The stream IDs for the units in CP-110 are configured in firmware
and are added to Device Tree, to be used by the SMMU driver.

- SATA unit is represented by a single SMMU entry, so added
  a single entry for SATA unit, which in fact represents
  both sata ports.

Change-Id: Ie78e9beb7ed2acbb3d8eab4d72804e093c2e8ef3
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38097
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index d95a3bd..37fd2fc 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -65,6 +65,7 @@
 			smmu: iommu@5000000 {
 				mmu-masters = <&usb3h0 0x483>,
 					      <&usb3h1 0x484>,
+					      <&sata 0x485>,
 					      <&cp_xor0 0x489>,
 					      <&cp_xor1 0x48a>;
 			};
@@ -119,6 +120,9 @@
 			ppv22@000000 {
 				dma-coherent;
 			};
+			sata: sata@540000 {
+				#stream-id-cells = <1>;
+			};
 			usb3h0: usb3@500000 {
 				#stream-id-cells = <1>;
 			};
-- 
1.7.9.5

